include/seqan3/io/detail/misc.hpp

```diff
@@ -78,14 +78,11 @@ void set_format(format_variant_type & format,
         extension = extension.substr(1); // drop leading "."
         meta::for_each(valid_formats{}, [&] (auto && fmt)
         {
-            using fmt_type = remove_cvref_t<decltype(fmt)>;
-            using fmt_tag  = typename fmt_type::format_tag;
-
-            for (auto const & ext : fmt_tag::file_extensions)
+            for (auto const & ext : fmt.file_extensions)
             {
                 if (std::ranges::equal(ext, extension))
                 {
-                    format = fmt_type{};
+                    format = std::move(fmt);
```

Comment:
Why this change?

Reply:
Since the `meta::for_each` now loops directly over the formats (and not over the `_REMOVEME` structs anymore), I can set the `format` function parameter by directly assigning `fmt`.

Reply:
But there is no state that you are preserving, right? To me the change implies that there is something in `fmt` that you want to store in `format` now. Or that it is prohibitively expensive to default-construct `fm_type{}` at that point...

Reply:
I wouldn't do it in that case, but I don't think we have strict rules on this or that you have to follow my advice on it.